WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -132,6 +132,11 @@ public async Task<Wallet> StartWalletAsync(Wallet wallet)
 
 			lock (Lock)
 			{
+				if (CancelAllInitialization.IsCancellationRequested || CancelAllInitialization.Token.IsCancellationRequested)
+				{
+					throw new OperationCanceledException($"Stopped loading {wallet}, because cancelling was requested.");
+				}
+
 				// Throw an exception if the wallet was not added to the WalletManager.
 				Wallets.Single(x => x == wallet);
 			}
```

Comment:
I believe it should be:

```cs
			if (CancelAllInitialization.IsCancellationRequested)
			{
				throw new OperationCanceledException($"Stopped loading {wallet}, because cancelling was requested.");
			}

			lock (Lock)
			{
				// Throw an exception if the wallet was not added to the WalletManager.
				Wallets.Single(x => x == wallet);
			}
```

given that `CancellationTokenSource` is thread-safe.

Reply:
But then `CancelAllInitialization` is canceled and disposed in `RemoveAndStopAllAsync` and we should not access disposed objects really. 

Imho the best fix would be not to call `StartWalletAsync` after `WalletManager` starts being disposed.

Reply:
> we should not access disposed objects really.

Damn, you are right. I will go with your approach.